test/prettier.js

```diff
@@ -124,29 +87,71 @@ ruleTester.run('prettier', rule, {
     '15',
     '16',
     '17',
-    '18'
-  ].map(loadInvalidFixture)
+    '18',
+  ].map(loadInvalidFixture),
 });
 
 const vueRuleTester = new RuleTester({
-  parser: require.resolve('vue-eslint-parser')
+  parser: require.resolve('vue-eslint-parser'),
 });
 
-vueRuleTester.run('prettier', rule, {
+vueRuleTester.run('vue', rule, {
   valid: [
     {
       code: `<template>\n  <div>HI</div>\n</template>\n<script>\n3;\n</script>\n`,
-      filename: 'valid.vue'
-    }
+      filename: 'valid.vue',
+    },
   ],
   invalid: [
     Object.assign(loadInvalidFixture('vue'), {
-      filename: 'invalid.vue'
+      filename: 'invalid.vue',
     }),
     Object.assign(loadInvalidFixture('vue-syntax-error'), {
-      filename: 'syntax-error.vue'
-    })
-  ]
+      filename: 'syntax-error.vue',
+    }),
+  ],
+});
+
+const atGraphqlEslintRuleTester = new RuleTester({
+  parser: require.resolve('@graphql-eslint/eslint-plugin'),
+});
+
+atGraphqlEslintRuleTester.run('@graphql-eslint/eslint-plugin', rule, {
+  valid: [
+    {
+      code: `type Query {\n  foo: String!\n}\n`,
+      filename: 'valid.graphql',
+    },
+  ],
+  invalid: [
+    Object.assign(loadInvalidFixture('graphql'), {
+      filename: 'invalid.graphql',
+    }),
+  ],
+});
+
+// eslint-plugin-graphql handles literal graphql files by tranforming graphql
+// code with a processor, instead of using a parser. Unfortunatly we cant
+// specify custom processors in a RuleTester, so instead we have write test code
+// that is the result of eslint-plugin-graphql's processing - this is the
+// ESLintPluginGraphQLFile tagged template literal. See
+// https://github.com/apollographql/eslint-plugin-graphql/blob/c465fedc8fea239ee1731ad4ec3ee1183a3cdddf/src/index.js#L404
+// In the future if ESLint supports processors (https://github.com/eslint/rfcs/pull/31)
+// we should be define a RuleTester like
+// `newRuleTester({processor: require('eslint-plugin-graphql').processor['.graphql']})
+// and then pass in pure graphql into the code value.
+const eslintPluginGraphqlRuleTester = new RuleTester({
+  parserOptions: { ecmaVersion: 2015 },
+});
+
+eslintPluginGraphqlRuleTester.run('eslint-plugin-graphql', rule, {
```

Comment:
Shouldn't `eslint-plugin-graphql` be a dependency too? 🤔

Reply:
No. This test does not run eslint-plugin-graphql currently, it merely simulates what that plugin's processors would output. As there it is not possible to run the processors from that plugin per the big comment above this line.